cornichon-core/src/main/scala/com/github/agourlay/cornichon/json/CornichonJson.scala

```diff
@@ -190,17 +192,58 @@ trait CornichonJson {
 
   implicit private val lcs: Patience[Json] = new Patience[Json]
 
-  def diffPatch(first: Json, second: Json): JsonPatch[Json] =
-    diff(first, second)
+  private def pathWithoutIndex(path: Pointer): Chain[Part] = {
+    path.parts.map {
+      case s @ Left(_) => s
+      case Right(_)    => Right(0)
+    }
+  }
+
+  def diffPatch(first: Json, second: Json, ignoreArrayOrdering: Boolean): JsonPatch[Json] = {
+    val allDiffs = diff(first, second)
+    if (!ignoreArrayOrdering) {
+      allDiffs
+    } else {
+      val eqJson: Eq[Json] = implicitly
+      val newDiffs: List[Operation[Json]] = allDiffs.ops.foldLeft(List.empty[Operation[Json]]) {
+        case (acc, e) =>
+          e match {
+            case a @ Add(path, value) =>
+              val aWithoutIndex = pathWithoutIndex(path)
+              println(aWithoutIndex)
+              val removed = allDiffs.ops.exists {
+                case Remove(path, rValue) if pathWithoutIndex(path) == aWithoutIndex && rValue.forall(v => eqJson.eqv(v, value)) => true
```

Comment:
Instead of computing the `pathWithoutIndex` each time, a possible optimization would be to do this computation once at the beginning. I don't know if it's really relevant or not.